src/libraries/System.Private.CoreLib/src/System/IO/Strategies/WindowsFileStreamStrategy.cs

```diff
@@ -120,6 +130,8 @@ internal sealed override SafeFileHandle SafeFileHandle
                     // in memory position is out-of-sync with the actual file position.
                     FileStreamHelpers.Seek(_fileHandle, _path, _filePosition, SeekOrigin.Begin);
                 }
+                _exposed = true;
+                _length = -1; // invalidate cached length
```

Comment:
We do always cache Length, but that's because FileStream always returns the proper length even if CanSeek is false.

Reply:
> but that's because FileStream always returns the proper length even if CanSeek is false

Can you elaborate? Length often isn't usable or meaningful if CanSeek is false.

Reply:
Ah right, I got confused since the CanSeek validation is not in the strategy but in the code of FileStream and thought Length was returning even if unseekable.

In that case, I think you are right and this could go inside above check.

Reply:
I could move it inside the `if` block but one day someone might need `_exposedHandle` for non-seekable file streams and then they would need to find out this subtle detail and move outside of the `if` block